dspace-api/src/main/java/org/dspace/util/SolrUpgradeStatistics6.java

```diff
@@ -0,0 +1,713 @@
+                        docs.add(input);
+                        ++numProcessed;
+                        if (numProcessed % batchSize == 0) {
+                                printTime(String.format("\t%,12d Processed...", numProcessed), false);
+                                batchUpdateStats();
+                                if (context.getCacheSize() > CACHE_LIMIT) {
+                                        refreshContext();
+                                }
+                        }
+                }
+                return numProcessed - initNumProcessed;
+        }
+        
+        
+        /*
+         * Map solr fields from legacy ids to UUIDs.
+         * 
+         * The id field is interpreted by the type field.
+         * The scopeId field is interpreted by scopeType field.
+         * 
+         * In order to prevent the re-processing of items with an id that cannot be resolved, the suffix "-legacy" will be appended to any id fields that cannot be resolved.  
+         * 
+         * @param input The SOLR statistics document to be updated
+         * @param col The SOLR field to update (if present)
+         */
+        private void mapField(SolrInputDocument input, FIELD col) throws SQLException {
+                SolrInputField ifield = input.get(col.name());
+                if (ifield != null) {
+                        Collection<Object> vals = ifield.getValues();
+                        ArrayList<String> newvals = new ArrayList<>();
+                        for(Object oval: vals) {
+                                try {
+                                        int legacy = Integer.parseInt(oval.toString());
+                                        UUID uuid = null;
+                                        if (col == FIELD.id) {
+                                                Object otype = input.getFieldValue("type");
+                                                if (otype != null) {
+                                                        int type = Integer.parseInt(otype.toString());
+                                                        uuid = mapType(type, legacy);
+                                                }
+                                        } else if (col == FIELD.scopeId) {
+                                                Object otype = input.getFieldValue("scopeType");
+                                                if (otype != null) {
+                                                        int type = Integer.parseInt(otype.toString());
+                                                        uuid = mapType(type, legacy);
+                                                }
+                                        } else {
+                                                uuid = mapId(col, legacy);
+                                        }
+                                        if (uuid != null) {
+                                                newvals.add(uuid.toString());
```

Comment:
Also, there is a UUIDField type in Solr, so this may also be a question whether it should be treated as a "string" or as a "uuid": https://wiki.apache.org/solr/UniqueKey#UUID_techniques